src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Windows.targets

```diff
@@ -84,19 +84,19 @@ The .NET Foundation licenses this file to you under the MIT license.
     </ItemGroup>
 
     <ItemGroup>
-      <LinkerArg Condition="'$(NativeLib)' == 'Shared'" Include="/DLL" />
+      <LinkerArg Condition="'$(NativeLib)' == 'Shared' or '$(CustomNativeMain)' == 'true'" Include="/DLL" />
```

Comment:
@jkotas question: are we sure we want to modify what `CustomNativeMain` would've normally done? Ie. I'm thinking, perhaps we should introduce another MSBuild property (eg. `CustomNativeMainOutputType` or whatever), which would allow users to choose whether they want to stick to the "normal" custom native main mode like on iOS, where it'd just change the name of the entry point of the .exe, or whether they want to actually publish a .dll, keep all the exports and everything, and just export this custom main as another function in this .dll? 🤔

Reply:
> they want to stick to the "normal" custom native main mode like on iOS

iOS SDK depends on IlcCompile target: https://github.com/xamarin/xamarin-macios/blob/38ef781bbe74f6c2ca5c5b1684ddb0dd2b1dbde5/dotnet/targets/Xamarin.Shared.Sdk.targets#L1438 . If somebody wants to run in iOS-like mode, they should do the same.

Reply:
We have https://github.com/dotnet/runtime/tree/main/src/tests/nativeaot/CustomMain test that got broken by these changes. (This test does not use the same flow as iOS SDK.) We will need to do something about the test or change how this is enabled. Changing how this is enabled may be better.

> CustomNativeMainOutputType or whatever

Maybe it can be the existing `<NativeLib>Shared</NativeLib>`?

Reply:
> "iOS SDK depends on IlcCompile target [...]"
Yeah no, I know, I just meant like, if someone in the future at some pointed wanted to use that base support here too, I was thinking perhaps it would be better to leave the default as one might expect by looking at how Xamarin uses it too, for consistency, and just rely on some other property in combination with `CustomNativeMain` (eg. `NativeLib` like you suggested) to achieve the exact behavior we want, rather than making `CustomNativeMain` do all the workl by itself and also making the assumption that you'd want a .dll as output necessarily.

> "Maybe it can be the existing `<NativeLib>Shared</NativeLib>`?"

That works! I made the changes in 853789cee52040c06976a8f90bd45f87805250d8, does that look ok?